lib/govuk_component/table_helpers.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+module GovukComponent
+  module TableHelpers
+    def govuk_table(**tag_options, &block)
+      tag_options = prepend_classes('govuk-table', tag_options)
+      table_block = capture(&block)
+
+      tag.table(table_block, tag_options)
+    end
+
+    def govuk_table_caption(**tag_options, &block)
+      tag_options = prepend_classes('govuk-table__caption', tag_options)
+      caption_block = capture(&block)
+
+      tag.caption(caption_block, tag_options)
+    end
+
+    def govuk_table_thead(**tag_options, &block)
+      tag_options = prepend_classes('govuk-table__head', tag_options)
+      thead_block = capture(&block)
+
+      tag.thead(tag_options) do
+        govuk_table_row do
+          thead_block
+        end
+      end
+    end
+
+    def govuk_table_tbody(**tag_options, &block)
+      tag_options = prepend_classes('govuk-table__body', tag_options)
+      tbody_block = capture(&block)
+
+      tag.tbody(tbody_block, tag_options)
+    end
+
+    def govuk_table_tfoot(**tag_options, &block)
+      tag_options = prepend_classes('govuk-table__foot', tag_options)
+      thead_block = capture(&block)
+
+      tag.tfoot(tag_options) do
+        govuk_table_row do
+          thead_block
+        end
+      end
+    end
+
+    def govuk_table_row(**tag_options, &block)
+      tag_options = prepend_classes('govuk-table__row', tag_options)
+      tr_block = capture(&block)
+
+      tag.tr(tr_block, tag_options)
+    end
+
+    def govuk_table_th(row = false, numeric = false, width = nil, **tag_options, &block)
```

Comment:
This is a case for keyword arguments if I ever saw one. I have identified three arguments (scope, numeric and width) for Govuk table header cell so far.
Keyword arguments makes it much more readable, `govuk_table_th(row: true, numeric: true, width: 'one-third')` instead of `govuk_table_th(true, true, 'one-third')`.

https://thoughtbot.com/blog/ruby-2-keyword-arguments

cc @jsugarman @kmahern

Reply:
yes in the sense of clarity but the number of arguments is very large. handling via `tag_options` is another way that is often used for view helpers in particular when args get unwieldy. i am not sure what is the "better" way

```
 def govuk_table_th(**tag_options, &block)
      klass = 'govuk-table__header'
      klass += ' govuk-table__header--numeric' if tag_options[:numeric]
      klass += " govuk-!-width-#{options[:width]}" if tag_options[:width]
      tag_options = prepend_classes(klass, tag_options)
      tag_options[:scope] = 'row' if tag_options[:row]
      ...
```

```
- tag_options = { row: '1|true|whatever'}
= govuk_table_th(tag_options) { t('shared.summary.expenses.travel_expenses') }
= govuk_table_td(tag_options) { claim.expenses_total }
```

there may be a case of having a different method for `row` or `col` if this is a common usage pattern - have been reading clean code and use of booleans to do different things in a method is discouraged generally, as is arguments exceeding 3